api/restapi/restapi.go

```diff
@@ -66,14 +67,39 @@ type peerAddBody struct {
 	PeerMultiaddr string `json:"peer_multiaddress"`
 }
 
+// copied from https://github.com/denji/golang-tls
+var tlsCfg = &tls.Config{
+	MinVersion:               tls.VersionTLS12,
+	CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
+	PreferServerCipherSuites: true,
+	CipherSuites: []uint16{
+		tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+		tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+		tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+		tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+	},
+}
+
 // NewRESTAPI creates a new REST API component. It receives
-// the multiaddress on which the API listens.
-func NewRESTAPI(apiMAddr ma.Multiaddr) (*RESTAPI, error) {
+// the multiaddress on which the API listens, as well as (potentially empty)
+// paths to certificate and private key files for TLS usage
+func NewRESTAPI(apiMAddr ma.Multiaddr, certFile, keyFile string) (*RESTAPI, error) {
```

Comment:
It's not a good practice to change the API (I mean, as in the signature of a public function of a module). So better to add a `NewTlsRESTAPI` or something like that.

A plus is that then you can make sure the user is not giving you an empty certificate path when he really means to use certificates.